client/src/App.js

```diff
@@ -13,67 +21,145 @@ function App() {
-    })
+    const updatedIngredients = toggleIngredient(ingredients, idToToggle)
     setIngredients(updatedIngredients)
   }
 
-  function filterActiveIngredients() {
-    const allActiveIngredients = ingredients.filter(ingredient => ingredient.isActive);
-    setActiveIngredients(allActiveIngredients)
+  useEffect(() => {
+    const allActiveIngredients = filterActiveIngredients(ingredients)
     saveToLocal('ingredients', ingredients)
     saveToLocal('activeIngredients', allActiveIngredients)
+  }, [ingredients])
+
+  const getRecipeResults = async () => {
+    setLoading(true)
+    const recipeData = await getRecipeData(activeIngredients, offsetCounter)
+    if (recipeData) {
+      setRecipes(recipeData)
+      saveToLocal('recipes', recipeData)
+      setLoading(false)
+      setError(false)
+    } else {
+      setLoading(false)
+      setError(true)
+    }
+  }
+
+  function deleteRecipe(idToDelete) {
+    const recipesToKeep = deleteItem(recipes, idToDelete)
+    setRecipes(recipesToKeep)
+    saveToLocal('recipes', recipesToKeep)
+  }
+
+  function addToLikedRecipes(recipeToAdd) {
+    const newRecipe = {
+      ...recipeToAdd,
+      isLiked: true
+    }
+
+    if (isNewEntry(likedRecipes, newRecipe)) {
+      setLikedRecipes([newRecipe, ...likedRecipes])
+    }
+
+    const allUnlikedRecipes = recipes.filter(recipe => recipe.id !== recipeToAdd.id);
+    setRecipes(allUnlikedRecipes)
+    saveToLocal('recipes', allUnlikedRecipes)
   }
 
   useEffect(() => {
-    filterActiveIngredients()
-  }, [ingredients])
+    saveToLocal('likedRecipes', likedRecipes)
+  }, [likedRecipes])
+
+
+  const increaseOffsetCounter = () => {
+    setOffsetCounter(offsetCounter + 6)
+    return offsetCounter
+  }
+
+  const getNextRecipeResults = async () => {
+    setLoading(true)
+    increaseOffsetCounter()
+    const nextRecipeData = await getRecipeData(activeIngredients, offsetCounter + 6)
+
+    if (nextRecipeData.length === 0) {
+      setLoading(false)
+      setError(true)
+      setOffsetCounter(0)
+    } else {
+      setLoading(false)
+      setRecipes(nextRecipeData)
+      saveToLocal('recipes', nextRecipeData)
+      setError(false)
+    }
+  }
 
   return (
-    <div className="App">
-      <Header
-        open={open}
-        setOpen={setOpen} />
-
-      <main>
-        <Switch>
-
-          <Route exact path="/">
-            <RecipeSearch
-              ingredients={ingredients}
-              onCreateIngredient={addIngredient}
-              onDeleteTag={deleteIngredient}
-              onToggleStatus={toggleActiveState} />
-          </Route>
-
-        </Switch>
-      </main>
-
-    </div>
+    <Router>
+      <div className="App">
+        <Header
+          open={open}
+          setOpen={setOpen} />
+
+        <main>
+          <Switch>
+
+            <Route exact path="/">
+              <RecipeSearch
+                ingredients={ingredients}
+                onGetRecipeResults={() => getRecipeResults()}
```

Comment:
I'm not sure if you need the arrow at the beginning? Might be the case that `onGetRecipeResults={getRecipeResults()}` works just as well.

Reply:
Thank you so much! You were right, it did work just fine without the arrow-function! :)